railties/test/application/test_runner_test.rb

```diff
@@ -651,6 +651,40 @@ class PostTest < ActiveSupport::TestCase
       end
     end
 
+    def test_declarative_style_regexp_filter_with_minitest_spec
+      app_file "test/models/post_test.rb", <<~RUBY
+        require "minitest/spec"
+
+        class PostTest < Minitest::Spec
+          it "greets foo" do
+            puts "hello foo"
+            assert true
+          end
+
+          it "greets foo again" do
```

Comment:
```suggestion
          it "greets  foo again" do
```

In my original test cases I had an extra space here because at one point my solution didn't work if there was anything more than a single space in the test name. Worth checking if that still works as expected here.

Reply:
Good point.  I've added an extra space to some test names in all three cases (string filter, regexp filter, and regexp filter with `Minitest::Spec`).

I've also added coverage for regexp special characters (e.g. `+`) in test names.